go-controller/pkg/ovn/policy.go

```diff
@@ -1320,8 +1320,9 @@ func (oc *Controller) addNetworkPolicy(policy *knet.NetworkPolicy) error {
 		if err := oc.deleteNetworkPolicy(policy, np); err != nil {
 			// rollback failed, add to retry to cleanup
 			key := getPolicyNamespacedName(policy)
-			oc.retryNetworkPolicies.initRetryObjWithDelete(policy, key, np, false)
-			oc.retryNetworkPolicies.unSkipRetryObj(key)
+			lockedKey := oc.retryNetworkPolicies.lockKey(key)
+			oc.retryNetworkPolicies.initRetryObjWithDelete(policy, lockedKey, np, false)
```

Comment:
Have a AddRetryObjectWithDelete in retryObjs that handles key locking internally